driver-core/src/main/com/mongodb/client/model/expressions/Expression.java

```diff
@@ -97,4 +97,24 @@ public interface Expression {
      * @return true if less than or equal to, false otherwise
      */
     BooleanExpression lte(Expression lte);
+
+    /**
+     * also checks for nulls
+     * @param or
+     * @return
+     */
+    BooleanExpression isBooleanOr(BooleanExpression or);
+    NumberExpression isNumberOr(NumberExpression or);
+    StringExpression isStringOr(StringExpression or);
+    DateExpression isDateOr(DateExpression or);
+    ArrayExpression<Expression> isArrayOr(ArrayExpression<? extends Expression> or);
+    <T extends DocumentExpression> T isDocumentOr(T or);
+
+
+    /**
+     * server error if type cannot be converted to string (arrays, objects)
```

Comment:
Remove this comment since it doesn't seem like what we will want in the end

Reply:
Please comment "Done" if this is done.

Reply:
Sorry, missed this - the latest variant of this comment (still in review in the docs pr #1059) is:

```
This may cause an error to be produced if the type cannot be converted
     * to a {@linkplain StringExpression string}, as is the case with
     * {@linkplain ArrayExpression arrays},
...
     
```